http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeServiceInterp.scala

```diff
@@ -403,12 +403,17 @@ class RuntimeServiceInterp[F[_]: Parallel](config: RuntimeServiceConfig,
 
       _ <- if (hasStopPermission) F.unit else F.raiseError[Unit](ForbiddenError(userInfo.userEmail))
       // throw 409 if the cluster is not stoppable
-      _ <- if (runtime.status.isStoppable) F.unit
+      _ <- if (runtime.status.isStoppable || runtime.status.isStopping) F.unit
       else
         F.raiseError[Unit](RuntimeCannotBeStoppedException(runtime.googleProject, runtime.runtimeName, runtime.status))
-      // stop the runtime
-      _ <- clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.PreStopping, ctx.now).transaction
-      _ <- publisherQueue.enqueue1(StopRuntimeMessage(runtime.id, Some(ctx.traceId)))
+
+      _ <- if (runtime.status.isStopping) F.unit
+      else {
+        // stop the runtime
+        clusterQuery.updateClusterStatus(runtime.id, RuntimeStatus.PreStopping, ctx.now).transaction
```

Comment:
This should be in a for-comp.. as is now, these are 2 un-chained `IO`s

Reply:
good catch!